### DISCUSSION TITLE
Proposal For Handling Hydraulic Inputs

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#301] [Q&A] [answered]
Proposing we handle hydraulic inputs in two forms.

As Raster Data - required
- Depth Grids or WSE Grids as .tifs or .vrts 
For this case, we use go-consequences to sample the points and generate the aggregated curves. I think this is necessary because this is a pretty typical, non RAS specific way to store hydraulic output. It also means the economists can add/move structures after the hydraulics have been developed without rerunning the shapefile through ras, as will be required for the next point. 

As Point Data - optional
I think we could work with the RAS team to implement something to attach WSE or depths to a point shapefile (the structure inventory). Then the hydraulic modeler could just hand over a point shapefile instead of a stack of data heavy depth grids. Alex K and I accomplished this with a console app once, but we could work to make a more official tool for RAS users. 
On our end, we'd ingest that shapefile, and connect those points to our structure inventory. At that point we could calculate the aggregated stage damage right there, or write up a method to do it in Go. I'm unsure if Go can accept point data as a hazard provider currently.  The upside of this method is that we don't have to move nearly as much data between people. For a big area, 8 depth grids could be 8gb. the same data expressed at depths on a shapefile would be something like 800kb.

**Comment by rnugent3** [comment]
I don’t think it is common for structures to get moved. If econs pass a refined structure inventory to engineering, the result should be good as long as the hydraulic modeling is good. Structure inventories _should_ get updated over the life of a study, but updates are often reflect structure characteristics or structure demolition.

**Comment by HenryGeorgist** [comment]
through go-inland we can get the depths at the points via go consequences from the hdf files directly, i think that would then service all major inputs from ras, hdf, tif, vrt.

**Comment by HenryGeorgist** [comment]
> I'm unsure if Go can accept point data as a hazard provider currently. The upside of this method is that we don't have to move nearly as much data between people. For a big area, 8 depth grids could be 8gb. the same data expressed at depths on a shapefile would be something like 800kb.
> 
This is a fine idea, we can easily have a hazard provider like this. This is similar to how I did go-fathom. FID for the structure and then return interval of the depths by column. We actually just used a csv file. Maybe we should go lo-tech like that.

**Comment by rnugent3** [comment]
Related question - do we need to handle the SID reference flood water surface profile? I am not sure I ever really understood what this thing is.

**Comment by Brennan1994** [comment]
Background information on how RAS output is stored is available here: https://github.com/HydrologicEngineeringCenter/HEC-FDA/discussions/308

My proposal above is not meant to require changes in RAS. I was considering building a separate console app we could host on github that could post process native RAS outputs into a single shapefile with all the required profiles appended. More and more though, I think our solution can be even simpler.

Have Hydraulics engineers provide their results as shapefiles appended to the structure inventory, hand over 8 point shapefiles either all in one, or in 8 subdirectories depending on whether they're from unsteady or steady computes. We do the join on our end, and create the input for Go Consequences to do the aggregation.

**Comment by rnugent3** [comment, marked as answer]
The team made a decision today to handle gridded data in the format of unsteady state RAS hydraulics gridded output as well as steady state hdf and unsteady hdf.
